cognition_objects/macro.py

```diff
@@ -0,0 +1,636 @@
+) -> List[CognitionMacro]:
+    project_item = project.get(project_id) if project_id else None
+    final_list = []
+    final_list = __get_admin_macros_for_me(
+        user, is_admin, project_item, only_production
+    )
+    final_list.extend(__get_org_macros_for_me(user, only_production))
+    if project_id:
+        final_list.extend(__get_project_macros_for_me(project_item, only_production))
+    return final_list
+
+
+def get_all_macro_executions(
+    macro_id: str,
+    execution_group_id: str,
+    state: Optional[Union[MacroExecutionState, List[MacroExecutionState]]] = None,
+) -> List[CognitionMacroExecution]:
+
+    query = session.query(CognitionMacroExecution).filter(
+        CognitionMacroExecution.macro_id == macro_id,
+        CognitionMacroExecution.execution_group_id == execution_group_id,
+    )
+    if state:
+        if is_list_like(state):
+            query = query.filter(
+                CognitionMacroExecution.state.in_([s.value for s in state])
+            )
+        else:
+            query = query.filter(CognitionMacroExecution.state == state.value)
+    return query.all()
+
+
+def get_macro_execution(
+    execution_id: str,
+    execution_group_id: str,
+    state: Optional[Union[MacroExecutionState, List[MacroExecutionState]]] = None,
+) -> CognitionMacroExecution:
+
+    query = session.query(CognitionMacroExecution).filter(
+        CognitionMacroExecution.id == execution_id,
+        CognitionMacroExecution.execution_group_id == execution_group_id,
+    )
+    if state:
+        if is_list_like(state):
+            query = query.filter(
+                CognitionMacroExecution.state.in_([s.value for s in state])
+            )
+        else:
+            query = query.filter(CognitionMacroExecution.state == state.value)
+    return query.first()
+
+
+def macro_execution_finished(
+    macro_id: str, execution_id: str, execution_group_id: str
+) -> bool:
+    return (
+        session.query(CognitionMacroExecution)
+        .filter(
+            CognitionMacroExecution.id == execution_id,
+            CognitionMacroExecution.macro_id == macro_id,
+            CognitionMacroExecution.execution_group_id == execution_group_id,
+            CognitionMacroExecution.state.in_(
+                [MacroExecutionState.CREATED.value, MacroExecutionState.RUNNING.value]
+            ),
+        )
+        .first()
+        is None
+    )
+
+
+def __get_admin_macros_for_me(
+    user: User, is_admin: bool, project: CognitionProject, only_production: bool
+) -> List[CognitionMacro]:
+
+    if (
+        not project
+        or not project.macro_config
+        or not (show := project.macro_config.get("show"))
+    ):
+        return []
+
+    if (
+        (show == AdminMacrosDisplay.DONT_SHOW.value)
+        or (show == AdminMacrosDisplay.FOR_ADMINS.value and not is_admin)
+        or (
+            show == AdminMacrosDisplay.FOR_ENGINEERS.value
+            and user.role != UserRoles.ENGINEER.value
```

Comment:
shouldnt admins still be possible to see this?

Reply:
hard to say since admin & role are in theory two distict values. not sure if the admin Annotator should be able to see them or not. I followed the suggestion and added the admin check.